spinn_front_end_common/abstract_models/impl/requires_tdma.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2020-2021 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from spinn_front_end_common.abstract_models.abstract_requires_tdma import \
+    AbstractRequiresTDMA
+from spinn_front_end_common.interface.provenance.\
+    provides_provenance_data_from_machine_impl import add_name
+from spinn_front_end_common.utilities import constants
+from spinn_front_end_common.utilities.utility_objs import ProvenanceDataItem
+from spinn_utilities.overrides import overrides
+
+
+# The number of clock cycles per nanosecond
+_CLOCKS_PER_NS = 200
+
+
+class RequiresTDMA(AbstractRequiresTDMA):
```

Comment:
This class has a slots problem. It needs its concrete subclasses (`AbstractPopulationVertex`, `SpikeSourcePoissonVertex`, and `DelayExtensionVertex`) to provide a very specific set of fields… and yet needs to have empty slots because of multiple inheritance. Something's very much not right. Perhaps making a `TDMAAwareApplicationVertex` (since all known uses are application vertices) instead of having this be totally separate would be a better approach?

Reply:
perfectly happy to do this if thats how we want to go.

Reply:
done